### DISCUSSION TITLE
".=" in variable assignment

### DISCUSSION THREAD

**Discussion by just-a-hriday** [dotnet/csharplang#5021] [Language Ideas] [answered]
It would be really nice if, instead of having to write this:
```c#
string foo;
foo = "hello World "; // note the extra space
foo = foo.ToLower(); // turns foo into "hello world "
foo = foo.Trim(); // turns foo into "hello world"
```
we could also write this:
```c#
string foo = "hello World "; // note the extra space
foo .= ToLower(); // should turn foo into "hello world "
foo .= Trim(); // should turn foo into "hello world"

//this should do the same thing.
```
This fits nicely with the fact that `foo += bar` is the same as `foo = foo + bar`, and `foo -= bar` is the same as `foo = foo - bar`. 

In case you didn't realize, I'm suggesting this for all types, not just strings.
Also, this should work for properties and fields too - not just functions. Here's an example with Where() being used on an array of doubles:
```c#
double[] doubles = new double[] { 12 , 34 , 16 , 0.5 , 0.25 , 65536 };

// print all the elements
foreach ( double num in doubles )
{
    Console.WriteLine(num);
}

Console.WriteLine();

// get all elements that are powers of two
doubles .= Where(num => ( Math.Log2(num) % 1 ) == 0).ToArray();

// print all the elements
foreach ( double num in doubles )
{
    Console.WriteLine(num);
}

//this should output:

//12
//34
//16
//0.5
//0.25
//65536

//16
//0.5
//0.25
//65536
```

I can't think of any examples for properties and fields right now.

**Comment by ufcpp** [comment, marked as answer]
https://github.com/dotnet/csharplang/issues/4253 https://github.com/dotnet/csharplang/discussions/4245
